src/main/java/core/basesyntax/strategy/impl/PurchaseHandlerImpl.java

```diff
@@ -1,24 +1,20 @@
 package core.basesyntax.strategy.impl;
 
 import core.basesyntax.dao.FruitDao;
-import core.basesyntax.dao.FruitDaoImpl;
 import core.basesyntax.model.Transaction;
 import core.basesyntax.strategy.OperationHandler;
 
 public class PurchaseHandlerImpl implements OperationHandler {
-    private final FruitDao fruitDao = new FruitDaoImpl();
+    private final FruitDao fruitDao;
+
+    public PurchaseHandlerImpl(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
 
     @Override
     public void applyChanges(Transaction transaction) {
-        if (fruitDao.get(transaction.getFruit()) == null) {
-            throw new RuntimeException("No fruit available: " + transaction.getFruit());
-        }
-        if (fruitDao.get(transaction.getFruit()) < transaction.getAmountFruits()) {
-            throw new RuntimeException("So many fruits are not available, only - "
-                    + fruitDao.get(transaction.getFruit()));
-        }
-        fruitDao.add(transaction.getFruit(),
-                fruitDao.get(transaction.getFruit())
-                        - transaction.getAmountFruits());
+        int remaining = fruitDao.getQuantity(transaction.getNameFruit());
+        int newQuantity = remaining - transaction.getAmount();
```

Comment:
what will be if remaining is less than `transaction.getAmount()`?
There should be some check